tick/eval/node_evaluator.go

```diff
@@ -0,0 +1,97 @@
+package eval
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/influxdata/kapacitor/tick"
+)
+
+// ErrTypeGuardFailed is returned when a speicifc value type is requested thorugh NodeEvaluator (for example: "Float64Value")
+// when the node doesn't support the given type, for example "Float64Value" is called on BoolNode
+type ErrTypeGuardFailed struct {
+	RequestedType ValueType
+	ActualType    ValueType
+}
+
+func CreateTypeGuardFailedError(requested, actual ValueType) error {
+	return &ErrTypeGuardFailed{requested, actual}
```

Comment:
Any reason we are using the pointer here? The `Error` method receiver is a value receiver so it shouldn't be necessary.